integration/container/mounts_linux_test.go

```diff
@@ -206,3 +211,58 @@ func TestMountDaemonRoot(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerBindMountNonRecursive(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux" || testEnv.IsRemoteDaemon())
```

Comment:
Could probably get rid of the remote daemon check if we do all the mounting in a container with shared propagation.

Reply:
Could you be more specific? Do you mean DIND?